lib/parallel_tests/cli.rb

```diff
@@ -2,6 +2,11 @@
 require 'tempfile'
 require 'parallel_tests'
 require 'shellwords'
+require_relative 'process_handler'
+
+# When this script receives the SIGINT signal from ctrl+c, send the SIGINT
+# signal to all child processes
+Signal.trap(:INT) { ParallelTests::ProcessHandler.kill_child_processes }
```

Comment:
this belongs into a separate PR ... I'm not even sure that is a real issue ... so far canceling always worked for me